src/modelinstance.cpp

```diff
@@ -305,9 +306,14 @@ static Status applyLayoutConfiguration(const ModelConfig& config, std::shared_pt
     }
 
     OV_LOGGER("ov::Model: {}, model->outputs()", reinterpret_cast<void*>(model.get()));
-    for (const ov::Output<ov::Node>& output : model->outputs()) {
+    size_t outputIndex = 0;
+    for (ov::Output<ov::Node>& output : model->outputs()) {
         try {
             OV_LOGGER("ov::Output<ov::Node> output: {}, output.get_any_name()", reinterpret_cast<const void*>(&output));
+            if (output.get_names().size() == 0) {
+                std::unordered_set<std::string> dummy_name{"OUT_" + std::to_string(outputIndex)};
+                output.add_names(dummy_name);
```

Comment:
This is not logic related to "applyLayoutConfiguration". Move those changes outside of this method.